pkg/deploy/lattice/service_network_synthesizer.go

```diff
@@ -117,7 +81,39 @@ func (s *serviceNetworkSynthesizer) synthesizeTriggeredGateways(ctx context.Cont
 	} else {
 		return nil
 	}
+}
+
+func (s *serviceNetworkSynthesizer) deleteServiceNetwork(ctx context.Context, resServiceNetwork *latticemodel.ServiceNetwork) string {
+	glog.V(6).Infof("Synthesizing Gateway: Del %v\n", resServiceNetwork.Spec.Name)
+
+	// TODO need to check if service network is referenced by gateway in other namespace
+	gwList := &gateway_api.GatewayList{}
+	s.Client.List(context.TODO(), gwList)
+	snUsedByGateway := false
+	for _, gw := range gwList.Items {
+		if gw.Name == resServiceNetwork.Spec.Name &&
+			gw.Namespace != resServiceNetwork.Spec.Namespace {
+			glog.V(6).Infof("Skip deleting gw %v, namespace %v, since it is still used", gw.Name, gw.Namespace)
+			snUsedByGateway = true
+			break
+		}
+	}
+
+	if snUsedByGateway {
+		glog.V(6).Infof("Skipping deleting gw: %v since it is still used by gateway(s)",
+			resServiceNetwork.Spec.Name)
+
+		return ""
+	}
+
+	err := s.serviceNetworkManager.Delete(ctx, resServiceNetwork.Spec.Name)
+	if err != nil {
+		return LATTICE_RETRY
```

Comment:
nit: return or log original error. For example:
```go
		return fmt.Errorf("%w: %w", RetryErr, err)
```
https://github.com/aws/aws-application-networking-k8s/blob/main/pkg/deploy/lattice/error.go#L13

Example: https://github.com/aws/aws-application-networking-k8s/blob/main/pkg/deploy/lattice/service_manager.go#L207-L214

Reply:
This was also a straight copy/paste/new method refactor, so for now I'll just add a log line. I'm hesitant to change too much of the flow logic at once.